addons/web_editor/models/media.py

```diff
@@ -0,0 +1,167 @@
+        self.ensure_one()
+        res_field = 'media_content' if self.media_content else None
+        return self.env['ir.attachment'].search([
+            ('res_field', '=', res_field),
+            ('res_id', '=', self.id),
+            ('res_model', '=', 'web_editor.media'),
+        ])
+
+    def _fetch_media_attachments(self, media_domain, attachments_domain, offset, limit, **kwargs):
+        """
+        Fetches the `ir_attachment`s linked to some `web_editor.media`.
+
+        :param media_domain: Search domain for the `web_editor.media` model.
+        :param attachments_domain: Search domain for the `ir_attachment` model.
+        :param int offset: Number of records to skip.
+        :param int limit: Maximum number of records to return.
+        :rtype: list(dict)
+        """
+        IrAttachment = self.env['ir.attachment']
+        fields = self._get_media_dialog_fields()
+        select_fields = (
+            [SQL.identifier(IrAttachment._table, field) for field in fields['attachment_fields']]
+            + [SQL.identifier(self._table, field) for field in fields['media_fields']]
+        )
+        media_where_clause = self._where_calc(media_domain).where_clause
+        attachments_where_clause = IrAttachment._where_calc(attachments_domain).where_clause
+
+        query = SQL("""
+            SELECT %(select_fields)s
+              FROM %(web_editor_media)s
+              JOIN "ir_attachment"
+                ON (
+                    "ir_attachment"."res_id"=%(web_editor_media)s.id
+                    AND "ir_attachment"."res_model"='web_editor.media'
+                )
+             WHERE %(media_where_clause)s
+                   AND %(attachments_where_clause)s
+          ORDER BY "ir_attachment"."id" DESC
+            OFFSET %(offset)s
+             LIMIT %(limit)s;
+        """,
+            select_fields=SQL(', ').join(select_fields),
+            web_editor_media=SQL.identifier(self._table),
+            media_where_clause=media_where_clause,
+            attachments_where_clause=attachments_where_clause,
+            offset=offset,
+            limit=limit,
+        )
+
+        self._cr.execute(query.code, query.params)
+        attachments = []
+        all_fields = fields['attachment_fields'] + fields['media_fields']
+        for result in self._cr.fetchall():
+            attachment = {all_fields[i]: result[i] for i in range(len(all_fields))}
+            # We can't directly query those fields in SQL, as they are computed.
+            # To avoid useless queries, we only fetch them if we have an image.
+            if attachment['mimetype'].split('/')[0] == 'image':
+                image_info = IrAttachment.search_read(
+                    [('id', '=', attachment['id'])],
+                    ['image_src', 'image_width', 'image_height'],
+                )[0]
+                attachment.update(image_info)
+            attachments.append(attachment)
+
+        return attachments
```

Comment:
I was not expecting a SQL query when making those specs:
- What justify it? And if it really is justified, we should add a comment about it in the docstring
- Why can't it just be a `search_read` done client-side (so no python code for this), which returns the bunch of web_editor.media ids you need? Do you actually need anything but the ids of the media when opening the media dialog?

Reply:
Same reason as the other explanations: `res_id` and `res_model`. I need the actual `res_id` / `res_model` (on the media record) but also a bunch of info on the attachment for the media dialog to treat the info. So I need a join on `ir.attachment` with a combined `where` clause.
JS-side, that would amount to at least 2 non-optimized requests:
- either I start by calling `ir.attachment`, but it cannot be optimized because how do I respect the limit while I don't know which attachments I should fetch for the current record?
- or I start with `web_editor.media`, but I don't have all the discriminating info, specifically the name and mimetype. So I end up with the same problem.

Reply:
Note that if I store an explicit relationship to the `attachment_id` as a field of `media`, I think there might be ways to simplify things with the ORM and maybe even manage it in one go from the JS (didn't test nor dug in that direction though, so I can't confirm for now).

Reply:
> That's what should be investigated before resorting to writing SQL queries

I confirm that there's no need for a SQL query if I store `media.attachment_id` and make a call with `.webSearchRead()`.

> Ideally, the media dialog should not know the ir.attachment model exists.

> The mimetype: should just not be needed. Why is it? 

That's not the only one, here are all the things that are stored on the attachment (not yet on the media) and retrieved for the dialog:

```js
'description',
'mimetype',
'checksum',
'type',
'access_token',
'original_id',
'image_src',
'image_width',
'image_height',
```

Some are then saved as data-attributes on the images, and then used in many different places in the codebase (both html_editor/web_editor & website). Either removing them from the media info (mimetype) or modifying their meaning (original_id as suggested in another of your comments) would have a lot of impacts. Are we sure that's what we want?

In addition, while some may be transferred to the media, a few are tied to the attachment itself: namely checksum, access_token, image_src / image_width / image_height (those 3 are computed, but image_src in particular is used in some places where a `media` record wouldn't be expected). We can discuss it, but I don't think we can do without them, meaning the subquery to `ir.attachment` needs to happen anyway?

Reply:
> That's not the only one

Problem is that I'll just ask the question for the rest of them then ^^ Without looking at the code, I really don't understand why the media dialog would want to know about any of those fields :grimacing: 

> Some are then saved as data-attributes on the images

Just the mimetype and the original_id no? But that won't be true anymore after LOCO's task... :grimacing: 

Probably worth speaking about it IRL, it will be easier (remind me maybe Wednesday?)